### PR TITLE
Reprocess SamplingProfiler entirely when loading a new module

### PR DESCRIPTION
**Pull request by dpallotti**
#### Remove SamplingProfiler::ProcessSamplesAsync
This is only used in Windows code that is currently unused and should be
implemented differently as it detaches a thread.
#### Add SamplingProfiler::ReprocessSamples to redo the entire processing
This replaces `SamplingProfiler::UpdateSampledFunctions` for use in
`SamplingReport::UpdateReport`. This is needed after loading a module as we
might have more information on addresses belonging to the same function.
Bug: http://b/157632145
#### Remove some old code from SamplingProfiler

### CODE REVIEW

File: OrbitCore/SamplingProfiler.cpp [hunk 1]
```diff
@@ -314,22 +316,42 @@ void SamplingProfiler::ProcessSamples() {
...
-          m_Process->GetFunctionFromAddress(sampled_function.m_Address, false);
-      if (function == nullptr) {
-        continue;
-      }
+void SamplingProfiler::ReprocessSamples() {
+  // Rebuild m_Callstacks (cleared by ProcessSamples) from m_ThreadSampleData.
+  // Note that m_Callstacks is not necessarily empty as more callstacks might
+  // have been added after the last call to ProcessSamples.
+  // Another option would be to just not call m_Callstacks.clear() in
+  // ProcessSamples.
```
[the remark this answers is not in the document]
Comment (dpallotti) [reply OrbitCore/SamplingProfiler.cpp]: Agree, so I removed this method and moved all the `clear()`s to `ProcessSamples`.

File: OrbitGl/SamplingReport.cpp [hunk 2]
```diff
@@ -43,7 +43,7 @@ void SamplingReport::FillReport() {
 }
 
 void SamplingReport::UpdateReport() {
```
Comment (dimitry-) [inline OrbitGl/SamplingReport.cpp:45]: The only question remains is that this method called twice on symbol update - when the request initiated and when we get data from the server. But I suppose this can be addresses in a separate PR.

Comment (dpallotti) [reply OrbitGl/SamplingReport.cpp:45]: Yeah, let's keep it for another change (it doesn't seem to cause any issues at the moment).
